spec/models/user_spec.rb

```diff
@@ -475,6 +475,22 @@ def assert_good(username)
       expect(User.username_available?('DonKey')).to eq(false)
       expect(User.username_available?('test')).to eq(false)
     end
+
+    it 'tests when a username is reserved by an expression with wildcard' do
+      SiteSetting.reserved_usernames = 'test)|*admin*|foo*|*bar|abc.def'
```

Comment:
This is also correct. I have updated that [here](https://github.com/discourse/discourse/pull/4728/files#diff-aeb8651199334c1cb831e884ff85773aR1043).